### ISSUE TITLE
'Precedence' header removal

### ISSUE THREAD

**Issue by SchupsCH** [ecederstrand/exchangelib#859]
Good day,

I am trying to remove the 'Precedence: bulk' header from incoming emails, so to allow their processing by a ticketing system.

```python
for index, header in enumerate(email.headers):
    if header.name == 'Precedence' and email.sender.email_address in whitelisted_email_addresses:
        email.headers.pop(index)
        break

email.save()
```

Unfortunately, on Exchange the email remains unchanged. This might be due to `headers` being `is_read_only=True` in `exchangelib/items/item.py`. If so, I am wondering if you could suggest any other way to achieve this?

Thank you!

**Comment by ecederstrand** [comment]
`headers` is a read-only field in EWS. I'm not sure, but it may be possible to manipulate the values using extended properties instead. See https://ecederstrand.github.io/exchangelib/#extended-properties

**Comment by SchupsCH** [comment]
Thank you, I tried with extended properties but I could not find a way to remove an already existing one.

When I am using the Outlook client, I am able to export the email to file, open with a text editor and remove the line of the header. Then I just drag the file back into Outlook, and deleting the original message. Tried approaching something similar with the to_xml and from_xml methods of Message, but having difficulties because I can't see what I am dealing with (can't view the data at hand). Do you suggest investing more time in this direction or is it a dead-end?

Thank you again!

**Comment by ecederstrand** [comment]
Ok, so you could doing the same in exchangelib. The equivalent is to fetch the item, create a new item by editing the contents of the `mime_content` field which should contain the header values, save the new item and delete the old item. You will have to experiment a bit which other fields to copy over from the old to the new item to make it look as much as the old items as necessary.

**Comment by ecederstrand** [comment]
Something like this would allow you to edit headers in the mime_content and replace the original item with a new one:

```python
item = a.inbox.all()[0]  # The item you want to replace
mime_parts = []
for l in item.mime_content.split(b'\n'):
    if l.startswith(b'Reply-To:'):
        l = b'Reply-To: Hello <mime@example.com>'
    mime_parts.append(l)
mime_content = b'\n'.join(mime_parts)
Message(folder=a.inbox, subject=item.subject, mime_content= mime_content).save()
item.delete()
```

**Comment by SchupsCH** [comment]
Thank you for coming back again on this. Much appreciated!

I attempted something similar and noticed something that worries me a bit (in short, the original mime_content was largely truncated). I will get back on this topic this week, and post back here my results.

**Comment by ecederstrand** [comment]
The `mime_content` field is probably sanitized so you can't spoof certain headers.

**Comment by SchupsCH** [comment]
```
new_email = Message(
    account=....,
    folder=...,
    mime_content=old_email.mime_content.replace(b'Precedence: bulk\r\n', b'')
)
new_email.save()
old_email.delete()
```

This is the only working way I was able to put together. I am a bit worried that the mime content get sanitized.

In any case we found a better solution that does not require replacing the email object, i.e. removing this header on a filter applied before the email reaches Exchange. So, I won't need this for the time being.

Thank you again for your support and quick answers
